src/Itertools.java

```diff
@@ -31,8 +31,14 @@ public Integer next() {
         return count(start, 1);
     }
 
-    public static <T> Iterable<T> cycle(List<T> list) {
+    public static <T> Iterable<T> cycle(Iterable<T> iterable) {
         return () -> new Iterator<T>() {
+
+		Iterator<T> iterator = iterable.iterator();
+
+			List<T> list = new ArrayList<>();
+			iterator.forEachRemaining(list::add);
```

Comment:
@Nitish145 This is rather a bad implementation as you are not utilising the iterable you are just storing them in a separate list and working from there on which defeats the purpose of supporting the iterables in the first place.